### ISSUE TITLE
`unicorn/require-post-message-target-origin` - typescript support

### ISSUE THREAD

**Issue by DetachHead** [oxc-project/oxc#24254]
[the documentation for this rule](https://oxc.rs/docs/guide/usage/linter/rules/unicorn/require-post-message-target-origin.html) says:

> Note that this rule may have false positives, as it is not capable of detecting all cases correctly without type information

since oxlint supports typescript, is there a reason this rule can't be updated to use type information?

related:

- https://github.com/oxc-project/oxc/issues/23922#issuecomment-4880025370
- #23918

**Comment by nrayburn-tech** [comment]
As of right now, type aware rules are only supported using tsgolint. Tsgolint is currently limited to only existing typescript-eslint rules.
https://github.com/oxc-project/tsgolint#status

That’s what was being alluded to in the closed issue that you linked.

**Comment by DetachHead** [comment]
i see, so is the plan to eventually expand type checking support to other rules once `tsgolint` is stable?

**Comment by afgomez** [comment]
Not sure if connected, but I'm getting a false positive when calling a function called `postMessage` in the slack SDK that is very clearly not the global one.

```
 × unicorn(require-post-message-target-origin): Missing the `targetOrigin` argument.
    ╭─[REDACTED]:10:48]
  9 │   try {
 10 │     return slackClient.chat.postMessage(message);
    ·                                                ▲
 11 │   } catch (error) {
    ╰────
  help: Insert `, self.location.origin`
  note: https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage
```

**Comment by camc314** [comment]
Hi, thanks for creating this issue.

Changing this behaviour at this time is not planned. We may revisit this in the future.

Thanks!
